examples/diabetes_example/run_diabetes.py

```diff
@@ -0,0 +1,500 @@
+# in the :ref:`API reference<documentation/api_reference:SDK Reference>` .
+#
+
+dataset = DatasetSpec(
+    name=f"Diabetes dataset",
+    type="csv",
+    data_opener=assets_directory / "dataset" / "diabetes_opener.py",
+    description=data_path / "description.md",
+    permissions=permissions,
+    logs_permission=permissions,
+)
+
+# We register the dataset for each of the organisations
+dataset_keys = {
+    client_id: clients[client_id].add_dataset(dataset)
+    for client_id in DATA_PROVIDER_ORGS_ID
+}
+
+for client_id, key in dataset_keys.items():
+    print(f"Dataset key {client_id} {key}")
+
+
+# %%
+# The dataset object itself is an empty shell. Data samples are needed in order to add actual data.
+# A data sample contains subfolders containing a single data file like a CSV and the key identifying
+# the dataset it is linked to.
+#
+
+datasample_keys = {
+    org_id: clients[org_id].add_data_sample(
+        DataSampleSpec(
+            data_manager_keys=[dataset_keys[org_id]],
+            test_only=False,
+            path=data_path / f"org_{i + 1}",
+        ),
+        local=True,
+    )
+    for i, org_id in enumerate(DATA_PROVIDER_ORGS_ID)
+}
+
+# %%
+# The data has now been added as an asset through the data samples.
+#
+
+# %%
+# Adding Functions
+# ================
+# A :ref:`Substra function<documentation/references/sdk_schemas:FunctionSpec>`
+# specifies the method to apply to a dataset or the method to aggregate models (artifacts).
+# Concretely, a function corresponds to an archive (tar or zip file) containing:
+#
+# - One or more Python scripts that implement the algorithm.
+# - A Dockerfile on which the user can specify the required dependencies of the Python scripts.
+#   This Dockerfile also specifies the method name to execute.
+#
+# In this example, we will:
+# 1. compute prerequisites for first-moment statistics on each data node;
+# 2. aggregate these values on the algo node to get aggregated statistics;
+# 3. send these aggregated values to the data nodes, in order to compute second-moment prerequisite values;
+# 4. finally, aggregate these values to get second-moment aggregated statistics.
+#
+
+
+# %%
+# First local step
+# ----------------
+# First, we will compute on each data node some aggregated values: number of samples, sum of each  numerical column
+# (it will be used to compute the mean), and count of each category for the categorical column (*Sex*).
+#
+# The computation is implemented in a *Python function* in the `federated_analytics_functions.py` file.
+# We also write a `Dockerfile` to define the entrypoint, and we wrap everything in a Substra `FunctionSpec` object.
+#
+# If you're running this example in a Notebook, you can uncomment and execute the next cell to see what code is executed
+# on each data node.
+
+# %%
+
+# %load -s local_step1 assets/functions/federated_analytics_functions.py
+
+# %%
+
+local_function_1_docker_files = [
```

Comment:
I think a better naming for functions could be useful for the user to understand what's happening. 
For instance, instead of `local_step1`, we could have smthing like `local_first_order_computation`, and rename  the other parameters accordingly. What do you think ?